### DISCUSSION TITLE
does it make sense to talk about rdf(graph)2rdf?

### DISCUSSION THREAD

**Discussion by majidaldo** [morph-kgc/morph-kgc#151] [General]
....or should i look into 'ontology alignmnt'?

in my case i want to 'homogenize' and stay in rdf for as much as possible (i have control over source data).

**Comment by arenas-guerrero-julian** [comment]
Hi @majidaldo ,

I assume that your input data source is an RDF KG, did I understood correctly?

In that case it would be possible, although we do not support RDF as input yet.

**Reply by majidaldo** [reply]
yes. how would that work?

**Comment by arenas-guerrero-julian** [comment]
You would define SPARQL queries targetting the input RDF, and then doing the data transformations in term maps to generate the output RDF.

**Reply by majidaldo** [reply]
> You would define SPARQL queries targetting the input RDF, and then doing the data transformations in term maps to generate the output RDF.

but i'm not seeing rdf as part of [rml](https://rml.io/specs/rml/)

**Reply by arenas-guerrero-julian** [reply]
RML can be extended to other data formats beyond those in the specification. For instance, [RMLMapper](https://github.com/RMLio/rmlmapper-java) already supports SPARQL endpoints as input.

**Reply by majidaldo** [reply]
ic. i was looking to use morphkgc though (should i look into extending it?). i want a python runtime. i'm also looking to use fno.

**Reply by arenas-guerrero-julian** [reply]
A release with RML+FnO support is planned for tomorrow :) . I do not plan to support SPARQL in the short term, but if you want to extend morph-kgc for SPARQL you can implement a new connector [here](https://github.com/morph-kgc/morph-kgc/tree/main/src/morph_kgc/data_source).
